packages/react/__tests__/rehydration.js

```diff
@@ -166,3 +197,131 @@ test('initializing another Emotion instance should not move already moved styles
     </head>
   `)
 })
+
+test('xxx', () => {
+  const { app, styles } = disableBrowserEnvTemporarily(() => {
+    resetAllModules()
+
+    let cache = createCache({ key: 'mui' })
+    let { extractCritical2, constructStyleTags } = createEmotionServer(cache)
+
+    const rendered = ReactDOMServer.renderToString(
+      <CacheProvider value={cache}>
+        <Global styles={{ body: { color: 'white' } }} />
+        <Global styles={{ html: { background: 'red' } }} />
+        <main css={{ color: 'green' }}>
+          <div css={{ color: 'hotpink' }} />
+        </main>
+      </CacheProvider>
+    )
+    const extracted = extractCritical2(rendered)
+    return {
+      app: extracted.html,
+      styles: constructStyleTags(extracted)
+    }
+  })
+
+  safeQuerySelector('head').innerHTML = styles
+  safeQuerySelector('body').innerHTML = `<div id="root">${app}</div>`
+  expect(safeQuerySelector('html')).toMatchInlineSnapshot(`
+    <html>
+      <head>
+        <style
+          data-emotion="mui-global l6h"
```

Comment:
this is somewhat surprising - I don't think I've ever seen such a short generated hash in Emotion but also while I was working on this I haven't really noticed anything wrong here and this seems to be OK, might still be worth checking this case in the `@emotion/hash`, just to be sure

Reply:
I don't think is really likely to have a collision, and I don't think it's worth even looking into changing the hashing algorithm based on one string that generates a shorter hash. I couldn't spot anything suspicious. It is funny on the other hand that it came up in styles we added in tests 😅

Reply:
Ye - nothing to worry about. I've still checked this out as I was curious.

At the very end of the hashing function:
https://github.com/emotion-js/emotion/blob/6c4a9e50f177900f54f7b3368a55b1a7d5e3c002/packages/hash/src/index.js#L63
we get `h` equal to `73383811246905` which in binary is `10000101011111000000000000000000110101100111001`. The first operation that is performed on this there is a bitwise right shift (`>>>`). Since this is 64bit number the left 32bit is just dropped, leaving us with `0000000000000000110101100111001` which only then gets shifted and as we shift by 15 positions we end up with `0`. XORing (`^`) the original `h` by `0` also discards the first 32 bit and only performs XOR after that which brings us to the `27449` result. So in a sense - stars just aligned nicely here and the left side of those 32 final bits are all `0`, leaving us with a small-ish result and thus the computed hash is short.